provisioning/transport/mqtt/src/ProvisioningChannelHandlerAdapter.cs

```diff
@@ -0,0 +1,423 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using DotNetty.Buffers;
+using DotNetty.Codecs.Mqtt.Packets;
+using DotNetty.Transport.Channels;
+using Microsoft.Azure.Devices.Provisioning.Client.Transport.Models;
+using Microsoft.Azure.Devices.Shared;
+using Newtonsoft.Json;
+using System;
+using System.Diagnostics;
+using System.Globalization;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Devices.Provisioning.Client.Transport
+{
+    internal class ProvisioningChannelHandlerAdapter : ChannelHandlerAdapter
```

Comment:
It normally takes a few seconds. However, it can be delayed for a variety of reasons - like IotHub not responding for example.

---
In reply to: [153912021](https://github.com/Azure/azure-iot-sdk-csharp/pull/248#discussion_r153912021) [](ancestors = 153912021)